### DISCUSSION TITLE
Symmetric docking with specific contacts

### DISCUSSION THREAD

**Discussion by Vini161** [RosettaCommons/rosetta#164] [Rosetta]
I have a hexamer ABCDEF, I want to make a 36 mer such that   CD stays out of it thats is QR interacts with AB-CD&EF-EF with GH-IJ&KL-KL with MN
I tried using C6_Z but it makes all units of hexamer contact everything.

<img width="430" alt="Screenshot 2024-09-17 at 6 14 21 PM" src="https://github.com/user-attachments/assets/8c13a8bd-1bcc-4a93-88b4-24f37fac5710">

example of how  36mer from hexamer should look like. Any guidance is appreciated

**Comment by roccomoretti** [comment]
If you have an example of the 36mer geometry, you should be able to use the [make-symmdef-file](https://docs.rosettacommons.org/docs/latest/application_documentation/utilities/make-symmdef-file) to encode the geometry of the symmetric 6-mer into the encoded symmetry. Note that this is different from the [make-symmdef-file-denovo](https://docs.rosettacommons.org/docs/latest/application_documentation/utilities/make-symmdef-file-denovo) script, which generates symmetries in the absence of an example structure. If you want more complex symmetries, you may need to encode them by hand (see [the symmetry documentation](https://docs.rosettacommons.org/docs/latest/rosetta_basics/structural_concepts/symmetry) for more.)

Note that for symmetric docking, part of the sampling process is the sampling of the DOFs between symmetric partners. In particular, it should be theoretically possible for the "monomer" (in your case the hetero-hexamer) to rotate freely around the center of mass. Symmetric docking is explicitly attempting to find the inter-subunit DOF values which best optimize the energy. So even if an abstract C6 symmetry doesn't initially generate the desired contacts, if you do symmetric docking properly, the docking process should be able to rotate the "monomer" to fix it.

That depends on the desired conformation being the lowest energy one. If your desired contacts aren't part of the lowest energy structure, then symmetric docking won't work properly. However, symmetric docking should be configurable with [constraints](https://docs.rosettacommons.org/docs/latest/rosetta_basics/file_types/constraint-file), so if you have contacts you want to enforce during docking, encoding them as constraints will cause the docking process to pull the docked structure into that form.
